install/kubernetes/cilium/values.yaml.tmpl

```diff
@@ -3178,6 +3178,12 @@ clustermesh:
             - ALL
       # -- lifecycle setting for the KVStoreMesh container
       lifecycle: {}
+      # -- Specify the KVStore mode when running KVStoreMesh
+      # Supported values:
+      # - "":         ``clustermesh-apiserver`` will use the containers ``apiserver``, ``etcd`` and ``kvstoremesh``
+      # - "external": ``clustermesh-apiserver`` will only use the container ``kvstoremesh`` and the user has to configure the etcd outside of the cluster, the KVStore address is set through `etcd.endpoints`.
```

Comment:
The`external` mode of operation can lead to surprising behavior if more than one replica of kvstoremesh is active at any time, because there would be two instances attempting to write the same keys to the same etcd cluster, given that it currently doesn't implement any leader election mechanism (because it is not needed with the current deployment). This problem could occur not only if explicitly configuring multiple replicas, but also during the rollout phase.

Reply:
We've talked about this on slack: https://cilium.slack.com/archives/C06AC2H4XQ9/p1730364907060869 . You stated that multiple replicas of clustermesh-apiserver (which includes kvstoremesh) are supported and recomended from 1.16.

Reply:
My statement in slack is true in the context of the currently supported deployment model, which creates a different stateless etcd instance for each replica of the clustermesh-apiserver (as I mentioned in the comment). 

However, it does not apply if kvstoremesh targets an external etcd cluster, because in that case you'd potentially end up with multiple replicas attempting to write the same keys to the same etcd cluster.